### PR TITLE
Update API and add integration testing.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: lib/src/_leak_checker.dart [hunk 1]
```diff
@@ -46,6 +46,9 @@ class LeakChecker {
   final LeakProvider leakProvider;
 
   void checkLeaks() {
+    // TODO: remove after debugging
+    print(StackTrace.current);
```
Comment (polina-c) [inline lib/src/_leak_checker.dart:50]: remove before merging

File: test/flutter/integration/README.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+This folder tests non-mocked public API of leak tracker.
+The tests can serve as exapmles for regression leak-testing for Flutter widgets.
```
Comment (CoderDake) [inline test/flutter/integration/README.md:2]:
```suggestion
The tests can serve as examples for regression leak-testing for Flutter widgets.
```

File: test/test_infra/helpers/gc.dart [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:developer';
+
+/// Forces garbage collection by aggressive memory allocation.
+Future<void> forceGC() async {
+  final _storage = <List<DateTime>>[];
+
+  void allocateMemory() {
+    _storage.add(Iterable.generate(10000, (_) => DateTime.now()).toList());
```
Comment (CoderDake) [inline test/test_infra/helpers/gc.dart:12]: If we are running these stress tests on our github bots, we may need to double check if we will be taxing any shared quotas.
Do you know if this is something that will take a lot of resources on the bots, or use a lot of time units?

File: lib/src/leak_tracker_model.dart [hunk 4]
```diff
@@ -16,6 +16,16 @@ class LeakTrackingConfiguration {
     this.classesToCollectStackTraceOnDisposal = const {},
   });
 
+  /// The leak tracker will not auto check leaks, and, when
+  /// leak checking is invoked, will notify only [listener].
```
Comment (CoderDake) [inline lib/src/leak_tracker_model.dart:20]:
```suggestion
  /// leak checking is invoked, will notify only [leakListener].
```

File: test/dart/platform_test.dart [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  return WeakReference(theObject);
+}
+
+late List<List<DateTime>> _storage;
+
+void _allocateMemory() {
+  _storage.add(Iterable.generate(10000, (_) => DateTime.now()).toList());
+}
+
+/// Tests for non-guaranteed asuumptions about Dart garbage collector,
```
Comment (CoderDake) [inline test/dart/platform_test.dart:28]:
```suggestion
/// Tests for non-guaranteed assumptions about Dart garbage collector,
```

**Review by CoderDake — APPROVED** [review]
LGTM, but I do wonder if the stress tests put any undue stress on the github bots, or our shared resources.
Do you know if that will be an issue?

File: test/test_infra/helpers/gc.dart (hunk 3 above)
Comment (polina-c) [reply test/test_infra/helpers/gc.dart:12]: If everything is as expected, this should not take a lot of resources: the cycle will stop at first garbage collection, that should happen much earlier than any significant impact for bots.

**Comment by polina-c** [comment]
> LGTM, but I do wonder if the stress tests put any undue stress on the github bots, or our shared resources. Do you know if that will be an issue?

If everything is as expected, this should not take noticeable resources: the cycle will stop at first garbage collection, that should happen much earlier than any significant impact for bots.

**Pull request merged**